wrappers/services.go

```diff
@@ -75,30 +76,109 @@ func generateSnapServiceFile(app *snap.AppInfo, opts *AddSnapServicesOptions) ([
 	return genServiceFile(app, opts)
 }
 
+// max returns the maximum of two integers. Why is this
+// not provided by golang?
+func max(a, b int) int {
+	if a < b {
+		return b
+	}
+	return a
+}
+
+func min(a, b int) int {
+	if b < a {
+		return b
+	}
+	return a
+}
+
+func formatCpuGroupSlice(grp *quota.Group) string {
+	if grp.CPULimit == nil {
+		return ""
+	}
+
+	// calculateSystemdCPULimit calculates the number of cpus and the allowed percentage
+	// to the systemd specific format. The CPUQuota setting is only available since systemd 213
+	calculateSystemdCPULimit := func() string {
```

Comment:
(nitpick^2) the helpers here feel a bit indirect, given that this feels relatively simple maybe a more direct approach similar to `formatMemoryGroupSlice` below like (comments still need tweaking).
```go
func formatCpuGroupSlice(grp *quota.Group) string {
	if grp.CPULimit == nil {
		return ""
	}

	header := `# Always enable cpu accounting, so the following cpu quota options have an effect
CPUAccounting=true
`
	// calculateSystemdCPULimit calculates the number of cpus and the allowed percentage
	// to the systemd specific format. The CPUQuota setting is only available since systemd 213
	buf := bytes.NewBufferString(header)
	if grp.CPULimit.Count != 0 || grp.CPULimit.Percentage != 0 {
		cpuQuotaSnap := max(grp.CPULimit.Count, 1) * grp.CPULimit.Percentage
		cpuQuotaMax := runtime.NumCPU() * 100
		fmt.Fprintf(buf, "CPUQuota=%d%%\n", min(cpuQuotaSnap, cpuQuotaMax))
	}
	// getAllowedCpusValue converts allowed CPUs array to a comma-delimited
	// string that systemd expects it to be in. If the array is empty then
	// an empty string is returned
	if len(grp.CPULimit.AllowedCPUs) > 0 {
		allowedCpusValue := strutil.IntsToCommaSeparated(grp.CPULimit.AllowedCPUs)
		fmt.Fprintf(buf, "AllowedCPUs=%s\n", allowedCpusValue)
	}
	buf.WriteString("\n")

	return buf.String()
}
```

Reply:
Yea totally see what you mean, the moment I figured golang supported lambda's I kinda went a bit crazy. It's a throwback to my c++ days. I love isolating functionality and lambda'ing them locally if they are only needed one place.